sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -251,64 +251,62 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
+      case table: FileTable =>
+        val write = table.newWriteBuilder(dsOptions).asInstanceOf[FileWriteBuilder]
+          .mode(modeForDSV1) // should not change default mode for file source.
+          .withQueryId(UUID.randomUUID().toString)
+          .withInputDataSchema(df.logicalPlan.schema)
+          .buildForBatch()
+        // The returned `Write` can be null, which indicates that we can skip writing.
+        if (write != null) {
+          runCommand(df.sparkSession, "save") {
+            WriteToDataSourceV2(write, df.logicalPlan)
+          }
+        }
+
+      case table: SupportsWrite if table.supports(BATCH_WRITE) =>
+        lazy val relation = DataSourceV2Relation.create(table, dsOptions)
+        modeForDSV2 match {
+          case SaveMode.Append =>
             runCommand(df.sparkSession, "save") {
-              WriteToDataSourceV2(write, df.logicalPlan)
+              AppendData.byName(relation, df.logicalPlan)
             }
-          }
 
-        case table: SupportsWrite if table.supports(BATCH_WRITE) =>
-          lazy val relation = DataSourceV2Relation.create(table, dsOptions)
-          modeForDSV2 match {
-            case SaveMode.Append =>
-              runCommand(df.sparkSession, "save") {
-                AppendData.byName(relation, df.logicalPlan)
-              }
-
-            case SaveMode.Overwrite if table.supportsAny(TRUNCATE, OVERWRITE_BY_FILTER) =>
-              // truncate the table
-              runCommand(df.sparkSession, "save") {
-                OverwriteByExpression.byName(relation, df.logicalPlan, Literal(true))
-              }
-
-            case other =>
-              throw new AnalysisException(s"TableProvider implementation $source cannot be " +
-                s"written with $other mode, please use Append or Overwrite " +
-                "modes instead.")
-          }
+          case SaveMode.Overwrite if table.supportsAny(TRUNCATE, OVERWRITE_BY_FILTER) =>
+            // truncate the table
+            runCommand(df.sparkSession, "save") {
+              OverwriteByExpression.byName(relation, df.logicalPlan, Literal(true))
+            }
 
-        // Streaming also uses the data source V2 API. So it may be that the data source implements
-        // v2, but has no v2 implementation for batch writes. In that case, we fall back to saving
-        // as though it's a V1 source.
-        case _ => saveToV1Source()
-      }
-    } else {
-      saveToV1Source()
+          case other =>
+            throw new AnalysisException(s"TableProvider implementation $source cannot be " +
+              s"written with $other mode, please use Append or Overwrite " +
+              "modes instead.")
+        }
+
+      // Streaming also uses the data source V2 API. So it may be that the data source
+      // implements v2, but has no v2 implementation for batch writes. In that case, we fall
+      // back to saving as though it's a V1 source.
```

Comment:
It doesn't look like this is a required change. It just changed the comment indentation and rewrapped. Can you remove this change to avoid future git conflicts?